splunk_add_on_ucc_framework/entity/__init__.py

```diff
@@ -13,19 +13,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-from typing import Dict, Any
+from typing import Dict, Any, List, Optional
 
 from splunk_add_on_ucc_framework.entity.entity import Entity
 from splunk_add_on_ucc_framework.entity.interval_entity import IntervalEntity
-
+from splunk_add_on_ucc_framework.entity.index_entity import IndexEntity
 
 ENTITY_TYPES = [
     IntervalEntity,
+    IndexEntity,
 ]
 
 
-def resolve_entity(entity_definition: Dict[str, Any]) -> Entity:
-    for entity_type in ENTITY_TYPES:
+def resolve_entity(
+    entity_definition: Dict[str, Any],
+    entity_types: Optional[List[Any]] = None,
```

Comment:
nitpick: can this be `Optional[List[Entity]]`?

Reply:
I think that it's because `ENTITY_TYPES` has no type set (line 22).